bindings/java/fdbJNI.cpp

```diff
@@ -305,6 +307,113 @@ JNIEXPORT jobject JNICALL Java_com_apple_foundationdb_FutureStrings_FutureString
 	return arr;
 }
 
+
+JNIEXPORT jobject JNICALL Java_com_apple_foundationdb_FutureKeyArray_FutureKeyArray_1get(JNIEnv *jenv, jobject, jlong future) {
+	if( !future ) {
+		throwParamNotNull(jenv);
+		return JNI_NULL;
+	}
+ 
+	FDBFuture *f = (FDBFuture *)future;
+
+	const FDBKey *ks;
+	int count;
+	fdb_error_t err = fdb_future_get_key_array( f, &ks, &count );
+	if( err ) {
+		safeThrow( jenv, getThrowable( jenv, err ) );
+		return JNI_NULL;
+	}
+
+	int totalKeySize = 0;
+	for(int i = 0; i < count; i++) {
+		totalKeySize += ks[i].key_length;
+	}
+
+	jbyteArray keyArray = jenv->NewByteArray(totalKeySize);
+	if( !keyArray ) {
+		if( !jenv->ExceptionOccurred() )
+			throwOutOfMem(jenv);
+		return JNI_NULL;
+	}
+	uint8_t *keys_barr = (uint8_t *)jenv->GetByteArrayElements(keyArray, JNI_NULL); 
+	if (!keys_barr) {
+		throwRuntimeEx( jenv, "Error getting handle to native resources" );
+		return JNI_NULL;
+	}
+
+	jintArray lengthArray = jenv->NewIntArray(count);
+	if( !lengthArray ) {
+		if( !jenv->ExceptionOccurred() )
+			throwOutOfMem(jenv);
+
+		jenv->ReleaseByteArrayElements(keyArray, (jbyte *)keys_barr, 0);
+		return JNI_NULL;
+	}
+
+	jint *length_barr = jenv->GetIntArrayElements(lengthArray, JNI_NULL); 
+	if( !length_barr ) {
+		if( !jenv->ExceptionOccurred() )
+			throwOutOfMem(jenv);
+
+		jenv->ReleaseByteArrayElements(keyArray, (jbyte *)keys_barr, 0);
+		return JNI_NULL;
+	}
+
+	int offset = 0;
+	for(int i = 0; i < count; i++) {
+		memcpy(keys_barr + offset, ks[i].key, ks[i].key_length);
+		length_barr[i] = ks[i].key_length;
+		offset += ks[i].key_length;
+	}
+
+	jenv->ReleaseByteArrayElements(keyArray, (jbyte *)keys_barr, 0);
+	jenv->ReleaseIntArrayElements(lengthArray, length_barr, 0);
+
+	jobject result = jenv->NewObject(key_array_result_class, key_array_result_init, keyArray, lengthArray);
+	if( jenv->ExceptionOccurred() )
+		return JNI_NULL;
+
+	return result;
+
+}
+
+JNIEXPORT jobject JNICALL Java_com_apple_foundationdb_FutureResults_FutureResults_1getSummary(JNIEnv *jenv, jobject, jlong future) {
```

Comment:
Is this function here by accident?

Reply:
Yes, due to switching of target branches.